sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,377 @@
+            raise TypeError(
+                f"Expected past_covariates to be a list, found {type(past_covariates)}."
+            )
+        self.past_covariates = past_covariates
+        if num_samples is not None and not isinstance(num_samples, int):
+            raise TypeError(
+                f"Expected num_samples to be an integer, found {type(num_samples)}."
+            )
+        self.num_samples = num_samples
+
+        super().__init__()
+
+        # initialize internal variables to avoid AttributeError
+        self._forecaster = None
+
+    @staticmethod
+    def convert_dataframe_to_timeseries(dataset: pd.DataFrame):
+        """Convert dataset for compatibility with ``darts``.
+
+        Parameters
+        ----------
+        dataset : pandas.DataFrame
+            source dataset to convert from
+
+        Returns
+        -------
+        darts.TimeSeries
+            converted target dataset
+        """
+        import darts
+
+        dataset_copy = _handle_input_index(dataset)
+        return darts.TimeSeries.from_dataframe(dataset_copy)
+
+    def convert_exogenous_dataset(
+        self: "_DartsAdapter", dataset: Optional[pd.DataFrame]
+    ):
+        """Make exogenous features to ``darts`` compatible, if available.
+
+        Parameters
+        ----------
+        dataset : Optional[pandas.DataFrame]
+            available data on exogenous features
+
+        Returns
+        -------
+        Tuple[darts.TimeSeries, darts.TimeSeries]
+            converted data on future known and future unknown exogenous features
+        """
+        if dataset is None and self.past_covariates:
+            raise ValueError(
+                f"Expected following exogenous features: {self.past_covariates}."
+            )
+
+        if dataset is None:
+            future_known_dataset = None
+            future_unknown_dataset = None
+
+        elif self.past_covariates is not None and self._lags_past_covariates:
+            future_unknown_dataset = self.convert_dataframe_to_timeseries(
+                dataset[self.past_covariates]
+            )
+            future_known_dataset = self.convert_dataframe_to_timeseries(
+                dataset.drop(columns=self.past_covariates)
+            )
+        elif self._lags_future_covariates:
+            future_unknown_dataset = None
+            future_known_dataset = self.convert_dataframe_to_timeseries(dataset)
+        else:
+            future_known_dataset = None
+            future_unknown_dataset = None
+
+        return future_known_dataset, future_unknown_dataset
+
+    @abc.abstractmethod
+    def _create_forecaster(self: "_DartsAdapter"):
+        """Create Darts model."""
+
+    @property
+    @abc.abstractmethod
+    def _lags_past_covariates(self):
+        """Get the lags_past_covariates value."""
+        pass
+
+    @property
+    @abc.abstractmethod
+    def _lags_future_covariates(self):
+        """Get the lags_future_covariates value."""
+        pass
```

Comment:
No, they don't all support past and future lags I think the better approach is to make these properties raise a `NotImplementedError` by default and when we implement them in the concrete classes this will be overridden, and introduce a property for each past and future lags that returns `None` when the `NotImplementedError` is raised.